Src/PerformanceCollector/Shared/Implementation/QuickPulse/QuickPulseCollectionStateManager.cs

```diff
@@ -125,8 +146,29 @@ public TimeSpan UpdateState(string instrumentationKey)
                     // no samples to submit, do nothing
                     return this.DetermineBackOffs();
                 }
+                else
+                {
+                    // we have samples
+                    if (dataSamplesToSubmit.Any(sample => sample.CollectionConfigurationAccumulator.GetRef() != 0))
+                    {
+                        ////!!! better solution?
+                        // some samples are still being processed, wait a little to give them a chance to finish
+                        Thread.Sleep(this.coolDownTimeout);
```

Comment:
Should we sleep and block the thread or have a Task.Delay or some other waiting solution instead?